lib/rspec/core/bisect/utilities.rb

```diff
@@ -32,8 +32,13 @@ def publish(event, *args)
       # parent process.
       # @private
       class Channel
+        MARSHAL_DUMP_ENCODING = Marshal.dump("").encoding
```

Comment:
Ah so, of course, `encoding` is not a method for 1.8.7, our usual work around would be to wrap it in a check, I'm about 80% certain 1.8.7s default encoding as just `:ascii` and in fact IO might not even have `set_encoding` which would mean you'd need to do a similar check there.
```suggestion
        if String.method_defined?(:encoding)
          MARSHAL_DUMP_ENCODING = Marshal.dump("").encoding
        else
          MARSHAL_DUMP_ENCODING = :ascii
        end
```

Reply:
Makes sense, thanks for the suggestion. I've confirmed that `IO#set_encoding` doesn't exist in 1.8.7, nor does the global `Encoding` object.

I've adjusted the code and specs to skip unsupported method calls, relying on whatever encoding is implicit to 1.8.7.

Hopefully this gets the PR across the line. I had trouble building MRI 1.8.7 on my end for testing so I'm relying on CI to reveal any issues. I'll address them as necessary.